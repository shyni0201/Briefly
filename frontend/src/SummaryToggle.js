import React from 'react';
import { ToggleButton, ToggleButtonGroup } from '@mui/material';
import { styled } from '@mui/material/styles';

/**
 * Styled component for custom toggle button appearance.
 * Extends MUI ToggleButton with custom colors and hover states.
 * 
 * @component
 * @param {object} theme - MUI theme object
 * @returns {Component} Styled ToggleButton component
 */
const CustomToggleButton = styled(ToggleButton)(({ theme }) => ({
  textAlign: 'center',
  flex: 0.18,
  '&.Mui-selected': {
    backgroundColor: '#1976D2',
    color: theme.palette.common.white, // White text for better contrast
    '&:hover': {
      backgroundColor: '#115293', // Darker shade for hover
    },
  },
  '&.Mui-selected:hover': {
    backgroundColor: '#115293', // Ensure hover color works when selected
  },
}));

/**
 * SummaryToggle Component - Toggles between personal and shared summaries view.
 * 
 * @component
 * @param {Object} props - Component props
 * @param {string} props.currentView - Current selected view ('my-summaries' or 'shared-summaries')
 * @param {function} props.onToggleChange - Callback function when toggle selection changes
 * @returns {JSX.Element} Toggle button group for switching between summary views
 */
const SummaryToggle = ({ currentView, onToggleChange }) => {
  /**
   * Handles toggle button selection change.
   * Only updates if a valid view is selected (prevents deselection).
   * 
   * @param {Event} event - The event object
   * @param {string} newView - The newly selected view value
   */
  const handleChange = (event, newView) => {
    if (newView !== null) {
      onToggleChange(newView);
    }
  };

  return (
    <ToggleButtonGroup
      value={currentView}
      exclusive
      onChange={handleChange}
      sx={{ mb: 3, display: 'flex'}}
    >
      <CustomToggleButton value="my-summaries">
        My Summaries
      </CustomToggleButton>
      <CustomToggleButton value="shared-summaries">
        Shared Summaries
      </CustomToggleButton>
    </ToggleButtonGroup>
  );
};

export default SummaryToggle;
