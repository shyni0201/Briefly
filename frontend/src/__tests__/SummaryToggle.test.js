import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom'; // Provides custom matchers like toBeInTheDocument()
import SummaryToggle from '../SummaryToggle';

// Mock any external dependencies if required
jest.mock('../RequestService', () => ({
  getUserSummaries: jest.fn(),
}));

describe('SummaryToggle Component', () => {
  let mockToggleChange;

  beforeEach(() => {
    mockToggleChange = jest.fn();
  });

  test('renders SummaryToggle with default state', () => {
    render(<SummaryToggle onToggleChange={mockToggleChange} />);

    // Verify buttons are rendered
    expect(screen.getByRole('button', { name: /my summaries/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /shared summaries/i })).toBeInTheDocument();

    // Verify default state
    expect(screen.getByRole('button', { name: /my summaries/i })).not.toHaveAttribute('aria-pressed', 'true');
    expect(screen.getByRole('button', { name: /shared summaries/i })).not.toHaveAttribute('aria-pressed', 'true');
  });

  test('calls onToggleChange when toggled to "My Summaries"', () => {
    render(<SummaryToggle onToggleChange={mockToggleChange} />);

    const mySummariesButton = screen.getByRole('button', { name: /my summaries/i });

    // Trigger the toggle action
    fireEvent.click(mySummariesButton);

    // Ensure callback is fired
    expect(mockToggleChange).toHaveBeenCalledWith('my-summaries');

    // Verify button states
    expect(mySummariesButton).toHaveAttribute('aria-pressed', 'false');
  });

  test('calls onToggleChange when toggled to "Shared Summaries"', () => {
    render(<SummaryToggle onToggleChange={mockToggleChange} />);

    const sharedSummariesButton = screen.getByRole('button', { name: /shared summaries/i });

    // Trigger the toggle action
    fireEvent.click(sharedSummariesButton);

    // Ensure callback is fired
    expect(mockToggleChange).toHaveBeenCalledWith('shared-summaries');

    // Verify button states
    expect(sharedSummariesButton).toHaveAttribute('aria-pressed', 'false');
  });


  test('matches snapshot for default state', () => {
    const { container } = render(<SummaryToggle onToggleChange={mockToggleChange} />);
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot for toggled state', () => {
    const { container } = render(<SummaryToggle onToggleChange={mockToggleChange} />);

    const mySummariesButton = screen.getByRole('button', { name: /my summaries/i });
    fireEvent.click(mySummariesButton);

    expect(container).toMatchSnapshot();
  });
});