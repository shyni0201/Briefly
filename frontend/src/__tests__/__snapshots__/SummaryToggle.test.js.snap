// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryToggle Component matches snapshot for default state 1`] = `
<div>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-951ssr-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-firstButton css-1y5lsam-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="my-summaries"
    >
      My Summaries
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-lastButton css-1y5lsam-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="shared-summaries"
    >
      Shared Summaries
    </button>
  </div>
</div>
`;

exports[`SummaryToggle Component matches snapshot for toggled state 1`] = `
<div>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-951ssr-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-firstButton css-1y5lsam-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="my-summaries"
    >
      My Summaries
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-lastButton css-1y5lsam-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="shared-summaries"
    >
      Shared Summaries
    </button>
  </div>
</div>
`;
